test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPublishTests.cs

```diff
@@ -2265,6 +2265,65 @@ public void PublishExcludeWithLongPath(string flavor, string os, string architec
             }
         }
 
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void PublishWithNoSourceOption_AppHasResourceFile(string flavor, string os, string architecture)
+        {
+            var runtimeHomeDir = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+
+            using (var tempDir = TestUtils.CreateTempDir())
+            {
+                var publishOutputPath = Path.Combine(tempDir, "output");
+                var appPath = Path.Combine(tempDir, "ResourcesTestProjects", "ReadFromResources");
+                TestUtils.CopyFolder(Path.Combine(TestUtils.GetMiscProjectsFolder(), "ResourcesTestProjects", "ReadFromResources"), appPath);
+                var workingDir = Path.Combine(appPath, "src", "ReadFromResources");
+
+                //Restore the application
+                var exitCode = DnuTestUtils.ExecDnu(runtimeHomeDir, "restore", appPath);
+                Assert.Equal(0, exitCode);
+
+                //Pack ResourcesLibrary project which will be used in publish below
+                exitCode = DnuTestUtils.ExecDnu(
+                    runtimeHomeDir,
+                    subcommand: "pack",
+                    arguments: "",
+                    environment: null,
+                    workingDir: Path.Combine(appPath, "src", "ResourcesLibrary"));
+                Assert.Equal(0, exitCode);
+
+                exitCode = DnuTestUtils.ExecDnu(
+                    runtimeHomeDir,
+                    subcommand: "publish",
+                    arguments: string.Format("--no-source --out {0}", publishOutputPath),
+                    environment: null,
+                    workingDir: workingDir);
+
+                Assert.Equal(0, exitCode);
+
+                var appOutputPath = Path.Combine(publishOutputPath, "approot", "packages", "ReadFromResources");
+                var versionDir = new DirectoryInfo(appOutputPath).GetDirectories().First().FullName;
+
+                Assert.True(File.Exists(Path.Combine(versionDir,
+                    "lib", "dnx451", "fr-FR", "ReadFromResources.resources.dll")), "Resources assembly did not get published for dnx451");
+                Assert.True(File.Exists(Path.Combine(versionDir,
+                    "lib", "dnxcore50", "fr-FR", "ReadFromResources.resources.dll")), "Resources assembly did not get published for dnxcore50");
+
+                appOutputPath = Path.Combine(publishOutputPath, "approot", "packages", "Microsoft.Data.Edm");
+                versionDir = new DirectoryInfo(appOutputPath).GetDirectories().First().FullName;
+                var edmLocales = new List<string>() {"de", "es", "fr", "it", "ja", "ko", "ru", "zh-Hans", "zh-Hant" };
```

Comment:
We need to make sure we're using a specific version of the package so that this doesn't vary over time.